components/percona_udf/percona_udf.cc

```diff
@@ -0,0 +1,76 @@
+/* Copyright (c) 2023 Percona LLC and/or its affiliates. All rights reserved.
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU General Public License
+   as published by the Free Software Foundation; version 2 of
+   the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA */
+
+#include <array>
+#include <bitset>
+
+#include <boost/preprocessor/stringize.hpp>
+
+#include <mysql/components/component_implementation.h>
+#include <mysqlpp/udf_registration.hpp>
+
+#include "percona_udf.h"
+
+// defined as a macro because needed both raw and stringized
+#define CURRENT_COMPONENT_NAME percona_udf
+#define CURRENT_COMPONENT_NAME_STR BOOST_PP_STRINGIZE(CURRENT_COMPONENT_NAME)
+
+REQUIRES_SERVICE_PLACEHOLDER(udf_registration);
+
+static const std::array known_udfs{DECLARE_UDF_INFO(fnv_64, INT_RESULT),
```

Comment:
May be, instead of adding empty `xxx_deinit()` functions define your own macro DECLARE_UDF_INFO_NO_DEINIT which would put `nullptr` for the `deinit` member.
```c++
#define DECLARE_UDF_INFO_NO_DEINIT(NAME, TYPE)                               \
  mysqlpp::udf_info {                                              \
    #NAME, TYPE, (Udf_func_any)&NAME, &NAME##_init, nullptr \
  }
```
or even use `mysqlpp::udf_info` directly
```c++
static const std::array known_udfs{
  mysqlpp::udf_info{"fnv_64", INT_RESULT, (Udf_func_any)&fnv_64, &fnv_64_init, nullptr},
  ...
```